Controller/CategoryAdminController.php

```diff
@@ -66,27 +66,33 @@ public function listAction(Request $request = null)
     public function treeAction(Request $request)
     {
         $categoryManager = $this->get('sonata.classification.manager.category');
-
-        $mainCategory = false;
         $currentContext = false;
         if ($context = $request->get('context')) {
             $currentContext = $this->get('sonata.classification.manager.context')->find($context);
         }
 
-        $rootCategories = $categoryManager->getRootCategories(false);
+        // all root categories
+        $rootCategories = $categoryManager->getRootCategoriesSplitByContexts(false);
+
+        // root categories inside the current context
+        $currentCategories = array();
 
         if (!$currentContext && !empty($rootCategories)) {
-            $mainCategory = current($rootCategories);
-            $currentContext = $mainCategory->getContext();
+            $currentCategories = current($rootCategories);
```

Comment:
@core23, mb I need to rename `rootCategories` variable, because it contains array of arrays (categories split by contexts), so `current` will return exactly what we need in this case (array of categories in some context)